pyOCD/target/cortex_m.py

```diff
@@ -488,7 +488,10 @@ def writeMemory(self, addr, value, transfer_size = 32):
         write a memory location.
         By default the transfer size is a word
         """
-        self.transport.writeMem(addr, value, transfer_size)
+        try:
+            self.transport.writeMem(addr, value, transfer_size)
+        except:
+            self.handleMemoryError()
```

Comment:
Since the target will be reset when a tap reset occurs, maybe it would make more sense to re-throw the exception and catch it at an even higher level.  If a reset occurs while running most scripts, the script will likely fail even if individual write functions don't throw an exception.  Scripts that need to be able to recover from a tap reset could be written like below:

```
flashing_done = False
while not flashing_done:
    try:
        flash.flashBinary(binary_file)
        flashing_done = True
    except ValueError: # or maybe TapResetError
        Pass
```